### ISSUE TITLE
C_Cpp.codeAnalysis.clangTidy.useBuildPath should not add "compile_commands.json" to the -p argument

### ISSUE THREAD

**Issue by sean-mcmanus** [microsoft/vscode-cpptools#9273]
Reading the other ticket again, I think this is related, but I agree it's not the same thing.  I can create another ticket for it if you like.

Now, to explain the issue I see with VSCode v1.66.2 with the C++ Extension v1.10:

This is what is being generated currently (again. from C/C++ logging window):
```
clang-tidy-13
--header-filter=/home/tmaslach/git
--quiet
/home/tmaslach/git/src/myfile.cpp
-p=/home/tmaslach/git/build/merged/compile_commands.json
```

This is what it should be generating:
```
clang-tidy-13
--header-filter=/home/tmaslach/git
--quiet
/home/tmaslach/git/src/myfile.cpp
-p=/home/tmaslach/git/build/merged
```

Notice the only thing removed is `compile_commands.json` in the last line.  Clang-tidy wants the folder compile_commands.json is in, not the actual file it points to.  It fails with that.

Thanks for the explanation in the last paragraph.  I do want clang-tidy to process the compile_commands.json file as the extension is not able to handle it properly. However, now that I know that when `useBuildPath` is `false`, you are trying to process the json file yourself, I will create another ticket once I have time explaining that issue.  I think it may be related to the fact that I have multiple `compile_commands.json` files in my build folder and I merge them together using cmake's setting: `cmake.mergedCompileCommands`

_Originally posted by @tmaslach in https://github.com/microsoft/vscode-cpptools/issues/8946#issuecomment-1117951525_

**Comment by sean-mcmanus** [comment]
@tmaslach Yeah, I remember hitting the `-p` not supposed to have the `compile_commands.json` previously -- should be easily fixable for our next release.

If you're using cmake, then installing the CMake Tools extension with "configurationProvider": "ms-vscode.cmake-tools" may be sufficient to get clang-tidy to be properly configured without using useBuildPath.

**Comment by sean-mcmanus** [comment]
@tmaslach FYI, I'm seeing clang-tidy handle -p with compile_commands.json at the end, but maybe there are some cases where it's not handled? We'll remove it in the next release anyway.

**Comment by tmaslach** [comment]
@sean-mcmanus  - What version of clang-tidy are you using?  Perhaps it's specific to version?  I was using 13.

**Comment by sean-mcmanus** [comment]
Yeah, I see the same behavior with clang-tidy 13 (on our build/test machines), i.e. same behavior with or without compile_commands.json in the -p=. However, may it's somehow auto-locating it depending on the relative path of the compile_commands.json.

**Comment by sean-mcmanus** [comment]
This is fixed with 1.10.2 (pre-release).

**Comment by tmaslach** [comment]
Works beautifully now, thank-you!
